src/app/files/remixDProvider.js

```diff
@@ -1,52 +1,45 @@
 'use strict'
 var EventManager = require('../../lib/events')
-var pathtool = require('path')
 
 module.exports = class RemixDProvider {
-  constructor (remixd) {
+  constructor (appManager) {
     this.event = new EventManager()
-    this._remixd = remixd
-    this.remixd = remixapi(remixd, this)
+    this._appManager = appManager
     this.type = 'localhost'
     this.error = { 'EEXIST': 'File already exists' }
     this._isReady = false
     this._readOnlyFiles = {}
     this._readOnlyMode = false
     this.filesContent = {}
     this.files = {}
+  }
 
+  _registerEvent () {
     var remixdEvents = ['connecting', 'connected', 'errored', 'closed']
     remixdEvents.forEach((value) => {
-      remixd.event.register(value, (event) => {
+      this._appManager.on('remixd', value, (event) => {
         this.event.trigger(value, [event])
       })
     })
 
-    remixd.event.register('notified', (data) => {
-      if (data.scope === 'sharedfolder') {
-        if (data.name === 'created') {
-          this.init(() => {
-            this.event.trigger('fileAdded', [this.type + '/' + data.value.path, data.value.isReadOnly, data.value.isFolder])
-          })
-        } else if (data.name === 'removed') {
-          this.init(() => {
-            this.event.trigger('fileRemoved', [this.type + '/' + data.value.path])
-          })
-        } else if (data.name === 'changed') {
-          this._remixd.call('sharedfolder', 'get', {path: data.value}, (error, content) => {
-            if (error) {
-              console.log(error)
-            } else {
-              var path = this.type + '/' + data.value
-              this.filesContent[path] = content
-              this.event.trigger('fileExternallyChanged', [path, content])
-            }
-          })
-        } else if (data.name === 'rootFolderChanged') {
-          // new path has been set, we should reset
-          this.event.trigger('folderAdded', [this.type + '/'])
-        }
-      }
+    this._appManager.on('remixd', 'folderAdded', (path) => {
+      this.event.trigger('folderAdded', [this.addPrefix(path)])
```

Comment:
Don't you need folder removed?

Reply:
file and folder removed share the same event `fileRemoved`